manifests/0000_07_cluster-network-operator_01_crd.yaml

```diff
@@ -67,14 +67,27 @@ spec:
               type: array
               items:
                 type: object
-                required: ["type", "name", "rawCNIConfig"]
+                required: ["type", "name"]
                 properties:
                   type:
                     type: string
                   name:
                     type: string
                   rawCNIConfig:
                     type: string
+                  macVlanConfig:
+                    type: object
+                    required: ["master", "ipam"]
+                    properties:
+                      master:
+                        type: string
```

Comment:
Is there any way we can auto-detect this? 

Given that we'll have to write a small daemonset that installs the macvlan CNI configuration, it can include a small binary that detects the uplink interface.

Reply:
In case of baremetal deployment, it is very hard I guess because user may wan to have several isolated networks over some hardware interfaces.

In addition, user may put hardware NICs asymmetry, such as:

- workerA: eth0(openshift-sdn), eth1(isolate netA)
- workerB: eth0(openshift-sdn),
- workerC: eth0(openshift-sdn), eth1(isolate netB), eth2(isolate netA)

And put the pod with pod affinity with their policy.

This case, we also need to discover isolated network as well as NIC somehow, so we could put it in future feature.

Reply:
While we definitely want people to be able to specify `master`, it seems like autodetecting and using the interface with the default route would be the right answer for a lot of users. It would be good if `master` wasn't required

Reply:
If it add macvlan to default route interface, this is the same network as openshift-sdn so it is not reason why they add additional interface. User want to add additional interface to connect other networks other than openshift-sdn, I believe. So I suppose we should have master parameter as required variable.

Reply:
In addition, macvlan to default route interface, the macvlan interface connects to host network (aka underlay network) and the interface will skip OvS flow rules. This seems to make the openshift network vulnerable. I am not clear on the use-case of such situation. What do you think about it?

Reply:
The use case is wanting a public IP distinct from the node's public IP. For instance, the OpenShift egress router uses a macvlan, but it is currently implemented via a special pod annotation that makes the openshift-sdn network plugin create the macvlan in addition to attaching the pod to the OVS bridge. In the future we want to implement it using multus.

Reply:
Also, putting a macvlan onto the same interface as openshift-sdn *does* mean increased performance because you're not running things through the VXLAN tunnel. So you can theoretically get closer to wirespeed on the macvlans. So it's still pretty useful for higher bandwidth pods even on the default interface.

Reply:
Okey, so 'master' goes to optional parameter.